### PR TITLE
burger menu button

### PR DESCRIPTION
**Pull request by harelglik**
Kapture 2021-04-18 at 01 55 11

### CODE REVIEW

File: packages/donor/src/components/AppHeader/AppHeader.tsx [hunk 1]
```diff
@@ -78,6 +93,25 @@ export default function ButtonAppBar({
...
+          close={() => setShowSideBar(false)}
+          navProps={{
+            navItemsProps: [
+              {
+                title: "החשבון שלי",
+                onClick: () => history.push("/" + MainNavigationKeys.MyProfile),
+              },
+              {
+                title: "התנתק",
+                onClick: () => firebase.auth().signOut(),
```
Comment (yaron1m) [inline packages/donor/src/components/AppHeader/AppHeader.tsx:108]: 1. Take signOut() as a prop here
2. Remove the sign out option from the profile page

File: packages/donor/src/components/AppHeader/AppHeader.tsx [hunk 2]
```diff
@@ -28,15 +33,18 @@ interface AppHeaderProps {
   title: string;
   hasBackButton?: boolean;
   hasProfileButton?: boolean;
+  hasBurgerMenu?: boolean;
```
Comment (yaron1m) [inline packages/donor/src/components/AppHeader/AppHeader.tsx:36]: Do you want to add a new story to the storybook?

File: packages/donor/src/components/AppHeader/AppHeader.tsx [hunk 3]
```diff
@@ -45,6 +53,13 @@ export default function ButtonAppBar({
     <AppBar position="fixed" color="secondary">
       <Toolbar>
         <Grid container>
+          <Grid item xs>
+            {hasBurgerMenu && (
+              <BurgerMenu
+                onClick={() => setShowSideBar((previous) => !previous)}
```
Comment (yaron1m) [inline packages/donor/src/components/AppHeader/AppHeader.tsx:59]: isnt this enough?
`onClick={() => setShowSideBar(!showSideBar)}`

File: packages/donor/src/components/AppHeader/AppHeader.tsx [hunk 4]
```diff
@@ -78,6 +93,25 @@ export default function ButtonAppBar({
           </Grid>
         </Grid>
       </Toolbar>
+      <div>
+        <SideDrawer
+          isOpen={showSideBar}
+          close={() => setShowSideBar(false)}
+          navProps={{
+            navItemsProps: [
```
Comment (yaron1m) [inline packages/donor/src/components/AppHeader/AppHeader.tsx:101]: Please add the about page (here or in another PR)

File: packages/donor/src/components/basic/ZMScreen/ZMScreen.tsx [hunk 5]
```diff
@@ -6,20 +6,20 @@ interface ExtendedSignupScreenProps {
...
   className = "",
   hasBackButton,
-  hasProfileButton,
+  hasBurgerMenu,
   title,
 }: ExtendedSignupScreenProps) {
   return (
     <>
-      <AppHeader {...{ hasBackButton, hasProfileButton, title }} />
+      <AppHeader {...{ hasBurgerMenu, hasBackButton, title }} />
```
Comment (yaron1m) [inline packages/donor/src/components/basic/ZMScreen/ZMScreen.tsx:22]: Should we somehow enforce we don't get both back button and burger menu?

File: packages/donor/src/components/AppHeader/AppHeader.tsx [hunk 6]
```diff
@@ -78,6 +93,25 @@ export default function ButtonAppBar({
...
+              {
+                title: "החשבון שלי",
```
Comment (yaron1m) [inline packages/donor/src/components/AppHeader/AppHeader.tsx:103]: The page title says פרופיל so let's be consistent

File: packages/donor/src/components/AppHeader/AppHeader.tsx (hunk 3 above)
Comment (harelglik) [reply packages/donor/src/components/AppHeader/AppHeader.tsx:59]: The correct way to update a state based on the previous state in react is to use the overloaded that gives you the previous state and not rely on the state value.

File: packages/donor/src/components/basic/ZMScreen/ZMScreen.tsx (hunk 5 above)
Comment (harelglik) [reply packages/donor/src/components/basic/ZMScreen/ZMScreen.tsx:22]: Don't think we should as in theory there is no reason why other pages should not be able to show both.

File: packages/donor/src/components/AppHeader/AppHeader.tsx (hunk 6 above)
Comment (harelglik) [reply packages/donor/src/components/AppHeader/AppHeader.tsx:103]: הפרופיל שלי

File: packages/donor/src/components/AppHeader/AppHeader.tsx (hunk 4 above)
Comment (harelglik) [reply packages/donor/src/components/AppHeader/AppHeader.tsx:101]: done

File: packages/donor/src/components/AppHeader/AppHeader.tsx (hunk 1 above)
Comment (harelglik) [reply packages/donor/src/components/AppHeader/AppHeader.tsx:108]: I started with the signout coming from outside but then you have to specify it and it's not always used to it becomes a little clumsy.
Is there a specific reason (testing?) why it's problematic to have it inlined here?

File: packages/donor/src/components/AppHeader/AppHeader.tsx (hunk 2 above)
Comment (harelglik) [reply packages/donor/src/components/AppHeader/AppHeader.tsx:36]: It's a bit trickier with the popups but I will try

File: packages/donor/src/components/AppHeader/AppHeader.tsx (hunk 1 above)
Comment (harelglik) [reply packages/donor/src/components/AppHeader/AppHeader.tsx:108]: 2 = done

**Review by yaron1m — APPROVED** [review]

**Pull request merged**